### ISSUE TITLE
Receiving Error message ITMS-90562: Invalid Bundle from Apple after submitting new binary for publication

### ISSUE THREAD

**Issue by ghb101** [RevenueCat/purchases-ios#1401]
I was using CocoaPods and updated to RevenueCat 4.1 from Purchases and then made the code adjustments and tested the app on my devices and all worked great. I submitted a binary to Apple and then received this message.

ITMS-90562: Invalid Bundle - One or more dynamic libraries that are referenced by your app are not present in the dylib search path.

I then decided to move from CocoaPods to Swift Package Manager (SPM) and after that transition while still using Purchases 3.x.x, and submitting the binary to Apple, all was good.  I then updated the SPM to RevenueCat 4.x.x + and all worked great on my devices.  But again, after submitting to Apple, I receive the same error message.

ITMS-90562: Invalid Bundle - One or more dynamic libraries that are referenced by your app are not present in the dylib search path.

I have RevenueCat used with just one target and I have checked that Purchases has been removed from the 'Link Binaries with Libraries' and RevenueCat appears there instead.

Any help will be appreciated.

thanks,
Glen

**Comment by ghb101** [comment]
1. The value for ALWAYS_EMEBED_SWIFT_STANDARD_LIBRARIES   is   $(EMBEDDED_CONTENT_CONTAINS_SWIFT). (Although another project I've tried this with, with the same error has it set to $(inherited). 
2. I have a Widget Extension, but not using RevenueCat in that at all
3. Yes, I am submitting with bitcode enabled

**Comment by ghb101** [comment]
I also did an ad-hoc export and pretty quickly this error message showed up.
ipatool failed with an exception: #<CmdSpec::NonZeroExitException: $ /Applications/Xcode.app/Contents/Developer/usr/bin/python3 /Applications/Xcode.app/Contents/Developer/usr/bin/bitcode-build-tool -v -t /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin --sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.4.sdk -o /var/folders/58/40rwgpjx04g588k7xctxmtr80000gp/T/ipatool20220318-40122-1rp6s7h/thinned-out/arm64/Payload/DailyTaskTracker.app/Frameworks/libswift_Concurrency.dylib --generate-dsym /var/folders/58/40rwgpjx04g588k7xctxmtr80000gp/T/ipatool20220318-40122-1rp6s7h/thinned-out/arm64/Payload/DailyTaskTracker.app/Frameworks/libswift_Concurrency.dylib.dSYM --strip-swift-symbols /var/folders/58/40rwgpjx04g588k7xctxmtr80000gp/T/ipatool20220318-40122-1rp6s7h/thinned-in/arm64/Payload/DailyTaskTracker.app/Frameworks/libswift_Concurrency.dylib
Status: pid 40154 exit 1
Stdout:
    SDK path: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/

**Comment by NachoSoto** [comment]
And I imagine your project is all Swift?
Can you send a screenshot of the list of targets? Do you have any frameworks to share code between your main app target and the widget extension?

I ask because it's possible that your widget extension is _implicitly_ importing `RevenueCat`, and therefore will need to have the dependency embedded in that target as well.

**Comment by NachoSoto** [comment]
Awesome, we're getting very close to figuring this out.

What's your app's deployment target?

**Comment by ghb101** [comment]
If I uncheck Rebuild from bitcode, then the ad hoc export works fine.
Using the latest Xcode 13.3
The iOS deployment target is 14.1 (Is that what you were asking for?)
I just use the Target Membership options to share some code between the main app and Widget.

I guess I could just add the RevenueCat dependency to the Widget and submit and see if that works.  Definitely wasn't needed with Purchases 3.x.x

**Comment by NachoSoto** [comment]
This was supposedly fixed in Xcode 13.3:

> Known Issues
Apps built with Xcode 13 or Xcode 13.1 that make use of Swift Concurrency features (such as async / await ), deploy to iOS prior to 15, tvOS prior to 15, or watchOS prior to 8, and have bitcode enabled may crash at launch with an error reporting that the libswift_Concurrency.dylib library was not loaded. (86349088) Workaround : Add -Wl,-weak-lswift_Concurrency -Wl,-rpath,/usr/lib/swift to Other Linker Flags in the app’s build settings.

**Comment by ghb101** [comment]
Interesting...I was just reading about this myself and wondering if Xcode 13 may be the issue.  
I guess I will just stick with Purchases 3.x.x for a while and try upgrade at a later date.  Thanks for all your help.
Maybe be good to put this warning in the docs as I doubt there are many apps only deploying to iOS 15, and Xcode 13 is about to be required for any app updates, so all developers will be using it if not already.

**Comment by NachoSoto** [comment]
I'm curious if that workaround would work for you? I imagine others might run into this with pre-iOS 15 deployment targets.

**Comment by ghb101** [comment]
Sadly, that workaround still gives the error.   : Add -Wl,-weak-lswift_Concurrency -Wl,-rpath,/usr/lib/swift to Other Linker Flags in the app’s build settings.

But I did update the Target deployment to iOS 15.0 (just to test) and that seems to work and not give the error. 
So here's hoping a future Xcode update solves the issue else I'll have to wait until after iOS 16 is released to think about updating the target deployment to 15.0

**Comment by NachoSoto** [comment]
I will lower the deployment target in my app and see if I can reproduce it and find a workaround.

**Comment by NachoSoto** [comment]
Sorry finally got around to taking a look at this. Turns out I've had my deployment target as 14.0 in my app for a while, and never saw this issue. I use Carthage but I've used SPM in the past too.

I would recommend filing a Radar attaching your project if you could so Apple can take a look.

**Comment by NachoSoto** [comment]
Looks like Xcode 13.3.1 has some fixes for Swift concurrency, would you be able to test if that fixes your issue?

**Comment by NachoSoto** [comment]
Actually I'm looking at the release notes and it seems like it solves exactly this issue:
> Fixed: Exporting an app that uses Swift’s concurrency features from an archive with bitcode might fail when the app targets iOS 13.0–14.7, watchOS 6.0–7.6, or tvOS 13.0–14.7. (89271047)

**Comment by ghb101** [comment]
Yes, thanks for the update.  I just updated Xcode to 13.3.1and generated an archive and exported for AdHoc, and it worked fine (that previously gave the error concurrency error).  I haven't tried uploading to AppStore, but it seems like that should work fine now.  I'll be able to test that in a day or so. Woo hoo....I'm so happy.

**Comment by NachoSoto** [comment]
Awesome! Let us know and we can close this once you verify it.

Glad you can upgrade to version 4 again!

**Comment by NachoSoto** [comment]
🎉 👏
